pkg/cli/initconfig/cmd/init.go

```diff
@@ -60,13 +59,25 @@ var initCmd = &cobra.Command{
 	Short: "init system config",
 	Long:  `init system config.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := run(); err != nil {
-			log.Fatal(err)
+		for {
+			time.Sleep(10 * time.Second)
+			if err := Healthz(); err != nil {
+				log.Error(err)
+				continue
+			}
+			if err := initSystemConfig(); err != nil {
+				log.Error(err)
+				break
+			}
+			log.Info("zadig init success")
 		}
 	},
 }
 
 func run() error {
```

Comment:
you'd better move all the logic above to this function, and simply call run() in `Run: func(cmd *cobra.Command, args []string) {}`